### ISSUE TITLE
Remove 4.2 build triggers and builds

### ISSUE THREAD

**Issue by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#1104]
## Notes about build candidate selection
- currently selecting candidates via `HAVING COUNT(*) < \(BuildPair.all.count)`  (where `BuildPair.all.count` = 40)
- if we were to delete from the builds table without server changes, we'd just re-trigger 4.2 builds
- if remove 4.2 from `BuildPair.all.count` we reduce it to 34 (4.2 has no ARM builds, hence 6 builds per version), which can affect candidate selection (although not in catastrophic ways)
	- new packages would be candidates, because 0 < 34 ✅
	- existing packages with gaps could be prevented from being candidates: say two 5.2 builds where missing, so 38 < 34 → false, i.e. no candidate, because the 4.2 builds pad out the missing builds ⚠️ (this will fix itself once we delete the 4.2 builds)
	- note that once a package is a candidate, we _do_ find the missing builds regardless of what versions are in `BuildPair.all.count`, because we compare all expected and existing versions (see `findMissingBuilds`) ✅

## TL;DR

- [x] 	remove 4.2 from `SwiftVersion`, reducing `BuildPair.all.count` to 34
- [x] 	deploy to stop triggering 4.2 builds
  - [x] dev
  - [x] prod
- [x] 	remove 4.2 builds from build table
  - [x] dev
  - [x] prod
- [x] remove 4.2 tests from spi-builder

```sql
delete from builds
where swift_version->'major' = '4' and swift_version->'minor' = '2'
```

**Comment by finestructure** [comment]
All done
